raft/src/main/java/org/apache/kafka/raft/internals/BatchAccumulator.java

```diff
@@ -147,6 +126,13 @@ private long append(int epoch, List<T> records, boolean isAtomic) {
 
         appendLock.lock();
         try {
+            long endOffset = nextOffset + records.size() - 1;
```

Comment:
I think must readers will assume that this "end offset" is exclusive. I think this offset is inclusive. We normally use `lastOffset` for this kind of offset.
```java
            long lastOffset = nextOffset + records.size() - 1;
```